core-tests/jvm/src/test/scala/zio/SemaphoreSpec.scala

```diff
@@ -3,58 +3,54 @@
 
 package zio
 
-import zio.duration._
-
 class SemaphoreSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRuntime {
 
   def is =
     "SemaphoreSpec".title ^ s2"""
     Make a Semaphore and
     verify that
-      `acquire` permits sequentially $e1
-      `acquire` permits in parallel $e2
-      `acquireN`s can be parallel with `releaseN`s $e3
-      individual `acquireN`s can be parallel with individual `releaseN`s $e4
+      `withPermit` performed sequentially $e1
+      `withPermit` performed in parallel $e2
+      multiple `withPermits` can be performed in sequence $e3
+      multiple `withPermits` can be performed in parallel $e4
       semaphores and fibers play ball together $e5
-      `acquire` doesn't leak permits upon cancellation $e6
-      `withPermit` does not leak fibers or permits upon cancellation $e7
-      `withPermitManaged` does not leak fibers or permits upon cancellation $e8
+      `withPermit` doesn't leak permits upon failure $e6
+      `withPermit` does not leak permits upon cancellation $e7
+      `withPermitManaged` does not leak permits upon cancellation $e8
     """
 
   def e1 = {
     val n = 20L
     unsafeRun(for {
       semaphore <- Semaphore.make(n)
-      available <- IO.foreach((0L until n).toList)(_ => semaphore.acquire) *> semaphore.available
-    } yield available must_=== 0)
+      available <- IO.foreach((0L until n).toList)(_ => semaphore.withPermit(semaphore.available))
```

Comment:
Makes me want to add a `semaphore.withAllPermits`

Reply:
... nothing to do with the test though, just thought that one might want a monopoly over all permits.

Reply:
Yes, if we went with this approach we could definitely maintain a `val` with the total number of permits and expose a `withAllPermits` method.